src/providers/aws.rs

```diff
@@ -554,14 +554,19 @@ where
                     return Ok(());
                 }
 
-                futures_util::future::join_all(self.regions.drain().map(|(region, mut rl)| {
-                    let region_span = tracing::debug_span!("region", %region);
-                    // this throws away a possible error from terminate_all,
-                    // but that is ok.
-                    async move { rl.terminate_all().await }.instrument(region_span)
-                }))
-                .await;
-                Ok(())
+                let res =
+                    futures_util::future::join_all(self.regions.drain().map(|(region, mut rl)| {
+                        let region_span = tracing::debug_span!("region", %region);
+                        async move { rl.terminate_all().await }.instrument(region_span)
+                    }))
+                    .await;
+                res.into_iter().fold(Ok(()), |acc, x| match acc {
+                    Ok(_) => x,
+                    Err(a) => match x {
+                        Ok(_) => Err(a),
+                        Err(e) => Err(a.wrap_err(e)),
```

Comment:
Yeah, this is pretty sad :'(
It's a same instance of the "list of errors" case I discussed with @yaahc in #32.
I think it's okay for now though.